### PR TITLE
Allow selecting region-of-interest in get_array()

### PR DESCRIPTION
**Pull request by takluyver**
Selecting a 2D region-of-interest would look something like this:

```python
run.get_array('SPB_IRU_INLINEMIC_CAM:daqOutput', 'data.image.pixels',
              roi=by_index[50:100, 20:250])
```

Slices are in the same order as dimensions in the HDF5 file, so probably [y, x] for camera images; this is normal for working with images in arrays. The syntax works for other numbers of dimensions, so you can also select e.g. a 1D region of interest in digitiser data.

Closes gh-100

### CODE REVIEW

File: karabo_data/reader.py [hunk 1]
```diff
@@ -589,14 +589,28 @@ def get_array(self, source, key, extra_dims=None):
...
+            holds a 1D array for each train, roi=by_index[:8] would get the
+            first 8 values from every train.
         """
         self._check_field(source, key)
+
+        if not isinstance(roi, by_index):
+            raise TypeError("roi parameter must be instance of by_index")
+        else:
+            roi = roi.value
+            if not isinstance(roi, tuple):
```
Comment (tmichela) [inline karabo_data/reader.py]: isnt' this always `False`?

**Comment by tmichela** [comment]
Maybe you could just update the doc-string describing how it works with multiple dimensions as you show in this PR description `by_index[a:b, c:d, e:f]`

**Comment by fangohr** [comment]
Nice. 

Can we have an example in the documentation, please? A notebook (or as part of one) would be preferred.

**Comment by takluyver** [comment]
See PR #111 for an example.

**Pull request merged**